src/Services/IAlgoliaIndexService.cs

```diff
@@ -7,10 +8,27 @@ namespace Kentico.Xperience.AlgoliaSearch.Services
     /// </summary>
     public interface IAlgoliaIndexService
     {
+        /// <summary>
+        /// Gets the desired name for the specified index. This can be used to append text to the index
+        /// names based on the current environment, e.g. "DEV-Xperience" and "PROD-Xperience."
+        /// </summary>
+        /// <param name="indexName">The original name of the index.</param>
+        /// <returns>The final name of the index.</returns>
+        string GetIndexName(string indexName);
+
+
         /// <summary>
         /// Initializes a new <see cref="ISearchIndex" /> for the given <paramref name="indexName" />.
         /// </summary>
         /// <param name="indexName">The code name of the index.</param>
         ISearchIndex InitializeIndex(string indexName);
+
+
+        /// <summary>
+        /// Sets the index's settings. See <see href="https://www.algolia.com/doc/api-reference/settings-api-parameters/"/>.
+        /// </summary>
+        /// <param name="indexName">The name of the index.</param>
+        /// <param name="indexSettings">The index settings.</param>
+        void SetIndexSettings(string indexName, IndexSettings indexSettings);
```

Comment:
I don't think that this is the right location for this method. This method is called by the system only within the registration phase so it should be designed just for that use case. The registration service seems much better for me.
Settings can be changed without this method in service...
`
var index = index.Service.InitializeIndex(..)
index.SetSettings(...)
`

Reply:
I removed this in https://github.com/Kentico/xperience-algolia/pull/24/commits/888a501ff25850ecc57517fdde90dfed3deed90d